src/mower_logic/src/mower_logic/mower_logic.cpp

```diff
@@ -483,6 +484,14 @@ void reconfigureCB(mower_logic::MowerLogicConfig &c, uint32_t level) {
     last_config = c;
 }
 
+bool startInAreaCommand(mower_msgs::StartInAreaSrvRequest &req, mower_msgs::StartInAreaSrvResponse &res) {
+    last_config.current_area = req.area;
+    last_config.clear_path_on_start = true;
+    if (currentBehavior) {
```

Comment:
Maybe could check if currentBehaviour is IdleBehavior and the call a method from there that sets up the MowingBehaviour to mow a single area and switches to it. Then I think the clear_path_on_start wouldn't  be required

Reply:
Actually, my first attempt was like that. I couldn't make it work, from what I remember the issue being clearing the state. So I went around with my limited understanding of the mower logic, most probably not optimal but compatible with the amount of time I could invest :-) .